### DISCUSSION TITLE
Possibility of resending email on forgot password expired action flow

### DISCUSSION THREAD

**Discussion by danieliu456** [keycloak/keycloak#23642] [Q&A]
I want to implement additional functionality regarding the forgot password email send (Resend email when action is expired) but can't figure out how to. I can see that UI sends unauthenticated requests with session data and tab ID, but maybe there is an endpoint so I can easily adjust the UI to add that button?

I can see from query params that ui has needed clientId and expired jwt token with UserId or email

Current flow:
User gets email -> link is expired -> end

Needed flow:
User gets email -> link is expired -> click resend email -> new email for user ... -> end.

**Comment by hisyamdarwis** [comment]
any update on this?

**Comment by seregacvetov62-design** [comment]
Also interested in
